corehq/apps/users/credentials_issuing.py

```diff
@@ -60,3 +67,92 @@ def get_app_ids_by_activity_level():
     for app in credential_apps:
         app_ids_by_level[app.activity_level].append(app.app_id)
     return app_ids_by_level
+
+
+def submit_new_credentials():
+    credentials_to_submit, credential_ids_to_update = get_credentials_to_submit()
+
+    response = requests.post(
+        settings.CONNECTID_CREDENTIALS_URL,
+        json={
+            "credentials": credentials_to_submit
+        },
+        auth=(settings.CONNECTID_CLIENT_ID, settings.CONNECTID_SECRET_KEY),
+    )
+    response.raise_for_status()
+
+    mark_credentials_as_issued(response, credential_ids_to_update)
+
+
+def get_credentials_to_submit():
+    from corehq.apps.users.models import ConnectIDUserLink, UserCredential
+
+    credentials_to_submit = []
+    credential_ids_to_update = []
+    user_credentials = UserCredential.objects.filter(issued_on=None)
+    if not user_credentials:
+        return credentials_to_submit, credential_ids_to_update
+
+    app_ids = []
+    usernames = []
+    for user_cred in user_credentials:
+        app_ids.append(user_cred.app_id)
+        usernames.append(user_cred.username)
+
+    connectid_links = ConnectIDUserLink.objects.filter(
+        commcare_user__username__in=usernames
+    )
+    connectid_username_by_commcare_username = {
+        link.commcare_user.username: link.connectid_username for link in connectid_links
+    }
+
+    app_names_by_id = get_app_names_by_id(app_ids)
+    for user_cred in user_credentials:
+        connectid_username = connectid_username_by_commcare_username.get(user_cred.username)
+        if not connectid_username:
+            continue  # Skip these users as they still need to set up their PersonalID account
+
+        credential_ids_to_update.append(user_cred.id)
+        credentials_to_submit.append({
+            'usernames': [connectid_username],
```

Comment:
This is always a list of one item. Can we combine multiple users that share a credential? Will it cause problems if we have the same credential multiple times in the same request but different sets of users?